blueprint/core/blueprint_EcmascriptEngine.cpp

```diff
@@ -228,6 +277,31 @@ namespace blueprint
         return result;
     }
 
+    //==============================================================================
+    void EcmascriptEngine::debuggerAttach()
+    {
+        DBG("Awaiting debugger attach...");
+
+        duk_trans_socket_init();
+        duk_trans_socket_waitconn();
+
+        duk_debugger_attach(ctx,
+                            duk_trans_socket_read_cb,
+                            duk_trans_socket_write_cb,
+                            duk_trans_socket_peek_cb,
+                            duk_trans_socket_read_flush_cb,
+                            duk_trans_socket_write_flush_cb,
+                            NULL,
+                            detail::handle_debugger_detached,
```

Comment:
For little functions like this I prefer inline lambdas (which cast to C-style function pointers as long as there's no capture group):

```
duk_debugger_attach(ctx,
    etc,
    etc,
    [](duk_context* ctx, void*) { duk_trans_socket_finish(); },
    NULL
);
```

Reply:
Well I just learnt a thing. Agreed.